src/devices/visonic_powercode.c

```diff
@@ -52,30 +52,50 @@ static int visonic_powercode_decode(r_device *decoder, bitbuffer_t *bitbuffer)
     uint8_t msg[32];
     data_t *data;
     char id[7];
+    uint8_t lrc_calculated = 0x00;
+    uint8_t lrc            = 0xff;
 
-    // 37 bits expected, 6 packet repetitions, accept 4 
+    // 37 bits expected, 6 packet repetitions, accept 4
     int row = bitbuffer_find_repeated_row(bitbuffer, 4, 37);
 
     // exit if anything other than one row returned (-1 if failed)
     if (row != 0)
         return DECODE_ABORT_LENGTH;
 
     // exit if incorrect number of bits in row or none
-    if (bitbuffer->bits_per_row[row] !=37)
+    if (bitbuffer->bits_per_row[row] != 37)
         return DECODE_ABORT_LENGTH;
 
     // extract message, drop leading start bit and trailing LRC nibble
-    bitbuffer_extract_bytes(bitbuffer, 0, 1, msg, 32);
+    bitbuffer_extract_bytes(bitbuffer, row, 1, msg, 32);
+
+    // extract LRC bits
+    bitbuffer_extract_bytes(bitbuffer, row, 33, &lrc, 4);
+    // Take high nibble of lrc
+    lrc            = (lrc & 0xf0) >> 4;
+    lrc_calculated = 0;
+    for (int i = 0; i < 4; i++) {
+        // low nibble
+        lrc_calculated ^= (msg[i] & 0xf);
+        // high nibble
+        lrc_calculated ^= ((msg[i] & 0xf0) >> 4);
+    }
+    lrc_calculated &= 0x0f;
```

Comment:
It's possibly easier to just xor the four bytes, then fold the nibbles once at the end. Also you can just add the chk too (expand the extract to 36) E.g. `b = xor_bytes(msg, 5); (b >> 4) ^ (b & 0xf) == 0`